File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prod = 1
        ifZero = 0
        for i in range(len(nums)):
            if nums[i] == 0:
                ifZero = ifZero + 1
                continue
            else:
                prod = prod * nums[i]
        
        res = []
        for i in range(len(nums)):
            if ifZero > 1:
                res.append(0)
            elif ifZero == 1:
                if nums[i] != 0:
                    res.append(0)
                else:
                    res.append(prod)
            else:
                res.append(prod//nums[i])
        
        return res

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();

        if(n<=1){
            return n;
        }

        int idx = 0;
        for(int i = 0; i< nums.size();i++){
            if(nums[i] != nums[idx]){
                nums[idx+1] = nums[i];
                idx++;
            }
        }

        return idx+1;
    }
};